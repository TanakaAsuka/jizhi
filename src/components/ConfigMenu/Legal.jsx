import React from 'react'
import { Text } from 'evergreen-ui'

const Legal = () => (
  <div style={{ height: 50 }}>
    <Text margin={20} lineHeight='50px' color='muted'>
      本扩展使用了
      <a href='https://www.jinrishici.com/' target='_blank' rel='noopener noreferrer'>
        今日诗词 API
      </a>
      ，感谢您的支持和喜爱!
    </Text>
  </div>
)

export default Legal
