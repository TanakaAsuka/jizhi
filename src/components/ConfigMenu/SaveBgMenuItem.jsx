import React, { Component } from 'react'
import { Menu } from 'evergreen-ui'
import { saveBackground } from '../../utils/'

export default class SaveBgMenuItem extends Component {
  render () {
    return (
      <Menu.Item
        icon='download'
        intent='success'
        onSelect={() => saveBackground()}
        secondaryText='Alt + S'
      >
        保存背景
      </Menu.Item>
    )
  }
}
