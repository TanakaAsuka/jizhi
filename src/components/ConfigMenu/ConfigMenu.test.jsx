import React from 'react'
import { shallow } from 'enzyme'

import ConfigMenu from '.'

describe('ConfigMenu', () => {
  const setupProps = {
    onSaveSelect: jest.fn(),
    onPlayPauseSelect: jest.fn(),
    isPlaying: true,
    defaultPlayChecked: true,
    onDefaultPlayChange: jest.fn(),
    colorStayChecked: false,
    onColorStayChange: jest.fn(),
    selected: 'test',
    onBgOptionChange: jest.fn(),
    isVerticalVerses: true,
    onVersesLayoutChange: jest.fn()
  }

  it('should render correctly', () => {
    const wrapper = shallow(<ConfigMenu {...setupProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should set isOpen state to true when calling onOpen function in Popover', () => {
    const wrapper = shallow(<ConfigMenu {...setupProps} />)
    wrapper.find('Popover').props().onOpen()

    expect(wrapper.state().isOpen).toEqual(true)
  })

  it('should set isOpen state to false when calling onClose function in Popover', () => {
    const wrapper = shallow(<ConfigMenu {...setupProps} />)
    wrapper.find('Popover').props().onClose()

    expect(wrapper.state().isOpen).toEqual(false)
  })
})
