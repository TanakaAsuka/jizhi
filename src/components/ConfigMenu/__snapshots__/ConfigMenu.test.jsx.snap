// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigMenu should render correctly 1`] = `ShallowWrapper {}`;
